marathon.go

```diff
@@ -113,22 +113,38 @@ func (provider *MarathonProvider) loadMarathonConfig() *Configuration {
 
 	//filter tasks
 	filteredTasks := fun.Filter(func(task marathon.Task) bool {
-		if len(task.Ports) == 0 {
-			log.Debug("Filtering marathon task without port", task.AppID)
-			return false
-		}
 		application, errApp := getApplication(task, applications.Apps)
 		if errApp != nil {
 			log.Errorf("Unable to get marathon application from task %s", task.AppID)
 			return false
 		}
+		//filter healthchecks
+		if application.HasHealthChecks() {
+			if task.HasHealthCheckResults() {
+				for _, healthcheck := range task.HealthCheckResult {
+					// found one bad healthcheck, return false
+					if !healthcheck.Alive {
+						log.Debugf("Filtering marathon task %s with bad healthcheck", task.AppID)
+						return false
+					}
+				}
+			} else {
+				log.Debugf("Filtering marathon task %s with bad healthcheck", task.AppID)
+				return false
+			}
+		}
+
+		if len(task.Ports) == 0 {
```

Comment:
Question : why moving this down ? If there is no ports, we don't even need to do a health check right ?.